homeassistant/components/vesync/light.py

```diff
@@ -51,34 +57,152 @@ def _async_setup_entities(devices, async_add_entities):
             # convert to percent that vesync api expects
             brightness = round((brightness / 255) * 100)
-            # clamp to 1-100
+            # ensure value between 1-100
             brightness = max(1, min(brightness, 100))
-            self.dimmer.set_brightness(brightness)
-        # Avoid turning device back on if this is just a brightness adjustment
-        if not self.is_on:
-            self.device.turn_on()
+            # call pyvesync library api method to set brightness
+            self.device.set_brightness(brightness)
+            # flag attribute_adjustment_only, so it doesn't turn_on the device redundantly
+            attribute_adjustment_only = True
+        # check flag if should skip sending the turn_on command
+        if attribute_adjustment_only:
+            return
+        # send turn_on command to pyvesync api
+        self.device.turn_on()
+
+
+class VeSyncDimmableLightHA(VeSyncBaseLight, LightEntity):
+    """Representation of a VeSync dimmable light device."""
+
+    def __init__(self, device):
+        """Initialize the VeSync dimmable light device."""
+        super().__init__(device)
+        self.device = device
 
     @property
-    def supported_features(self):
-        """Get supported features for this entity."""
-        return SUPPORT_BRIGHTNESS
+    def color_mode(self):
+        """Set color mode for this entity."""
+        return COLOR_MODE_BRIGHTNESS
 
     @property
-    def brightness(self):
-        """Get dimmer brightness."""
-        return round((int(self.dimmer.brightness) / 100) * 255)
+    def supported_color_modes(self):
+        """Flag supported color_modes (in an array format)."""
+        return [COLOR_MODE_BRIGHTNESS]
+
+
+class VeSyncTunableWhiteLightHA(VeSyncBaseLight, LightEntity):
+    """Representation of a VeSync Tunable White Light device."""
+
+    def __init__(self, device):
+        """Initialize the VeSync Tunable White Light device."""
+        super().__init__(device)
+        self.device = device
+
+    @property
+    def color_temp(self):
+        """Get device white temperature."""
+        # get value from pyvesync library api,
+        result = self.device.color_temp_pct
+        try:
+            # check for validity of brightness value received
+            color_temp_value = int(result)
+            # flip cold/warm
+            color_temp_value = 100 - color_temp_value
+            # ensure value between 0-100
+            color_temp_value = max(0, min(color_temp_value, 100))
+            # convert percent value to Mireds
+            color_temp_value = round(
+                self.min_mireds
+                + ((self.max_mireds - self.min_mireds) / 100 * color_temp_value)
+            )
+            # ensure value between minimum and maximum Mireds
+            color_temp_value = max(
+                self.min_mireds, min(color_temp_value, self.max_mireds)
+            )
+            return color_temp_value
```

Comment:
Please put only the code that can throw an exception inside the try block.

We want to show the intent of the exception trap